blockchain/version.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2013-2016 The btcsuite developers
+// Copyright (c) 2015-2016 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package blockchain
+
+import "fmt"
+
+// calcNextStakeVersion calculates the next stake version based on the provided
+// node.  The next best stake version is defined as a 95% super majority of all
+// stake versions at a StakeDiffWindowSize interval.
+//
+// If the node is not at a StakeDiffWindowSize it returns the prior version.
+//
+// This function is safe for concurrent access.
+func (b *BlockChain) calcNextStakeVersion(node *blockNode) (uint32, error) {
```

Comment:
After reviewing this more closely, the overall methodology of this function doesn't really seem accurate to me and actually seems quite dangerous.  The way it is currently written would allow intentionally malformed votes to arbitrarily bump the version higher than the expected version.

The function should really just almost mirror the `isMajorityVersion` function and thus be `isMajorityStakeVersion(minVer int32, startNode *blockNode) (bool, error)` and it should just be tallying the number of votes with a `version >= minVer`.

Off the top of my head, it would be something like the following:
```Go
	var totalVotes, numFound uint64
	for i := uint64(0); i < b.chainParams.StakeDiffWindowSize; i++ {
		for _, v := range oldNode.voterVersions {
			if v > minVer {
				numFound++
			}
			totalVotes++
		}
		...
	}

	threshold := (totalVotes *
		b.chainParams.SuperMajorityMultiplier) / 
		b.chainParams.SuperMajorityDivisor
	return numFound >= threshold
```

There is no need to calculate this value.  Instead, miners should be creating their votes with the value configured by `b.chainParams.StakeVersion`.